controlplane/kubeadm/internal/workload_cluster_conditions.go

```diff
@@ -252,13 +255,15 @@ func (w *Workload) UpdateStaticPodConditions(ctx context.Context, controlPlane *
 	// NOTE: this fun uses control plane nodes from the workload cluster as a source of truth for the current state.
 	controlPlaneNodes, err := w.getControlPlaneNodes(ctx)
 	if err != nil {
+		log.Error(err, "Failed to retrieve control plane nodes")
+
 		for i := range controlPlane.Machines {
 			machine := controlPlane.Machines[i]
 			for _, condition := range allMachinePodConditions {
-				conditions.MarkUnknown(machine, condition, controlplanev1.PodInspectionFailedReason, "Failed to get the node which is hosting this component: %v", err)
+				conditions.MarkUnknown(machine, condition, controlplanev1.PodInspectionFailedReason, "Failed to get the node which is hosting this component")
```

Comment:
Why is this changing?

Reply:
Trying to answer all 3 comments here.

The problem is that exposing errors we don't control in conditions can lead to infinite reconciles. So we cannot include the error message in a condition. Because we also don't return the errors we are logging them to give folks a chance to debug issues.

Do you see a better alternative?

Reply:
@vincepri ^^